controllers/admin/AdminCartRulesController.php

```diff
@@ -274,6 +274,7 @@ public function processAdd()
             $this->redirect_after = false;
             if ($cart_rule) {
                 $this->context->smarty->assign('refresh_cart', true);
+                $this->context->smarty->assign('created_cart_rule', $cart_rule->id);
```

Comment:
```suggestion
                $this->context->smarty->assign('created_cart_rule_id', $cart_rule->id);
```

Reply:
I think both variables have the same effect actually So they should be merged into one You can replace `refresh_cart` with only `created_cart_rule_id` and in the template you can only check for `created_cart_rule_id` presence and call the `addCreatedCartRuleToCart` js function only

The internally `addCreatedCartRuleToCart` can call the `refreshCart` function this way you are sure, the add cart rule action has been performed fully before the cart is refreshed and it reduces the quantity of code necessary in the smarty template